src/plugins/intel_cpu/src/edge.cpp

```diff
@@ -52,22 +52,43 @@ bool Edge::isDropped() const {
 }
 
 void Edge::collectConsumers(std::vector<NodePtr>& result) const {
-    if (!this->getChild()->getChildEdges().empty() && this->inPlace(LOOK_DOWN)) {
-        if (auto peerChildSPD = this->getChild()->getSelectedPrimitiveDescriptor()) {
+    auto childNode = this->getChild();
+    if (childNode->getChildEdges().empty()) {
+        if (Type::ShapeOf != childNode->getType()) {
+            // ShapeOf doesn't actually read the data, it only reads shape
+            result.push_back(childNode);
+        }
+        return;
```

Comment:
Now we have two same condition checks on L57-61 and L74-78.
It looks like we need to merge these to checks in one lambda:
```cpp
auto add_result_node = [](std::vector<NodePtr>& result, const NodePtr& node) {
    if (Type::ShapeOf == node->getType()) {
        // ShapeOf doesn't actually read the data, it only reads shape
        return;
     }
     result.push_back(node);
}
```
And call it in those two places.

Reply:
done